.github/actions/create-jira-ticket/action.yml

```diff
@@ -0,0 +1,162 @@
+            "maxResults": 1
+        }')
+
+        if [ $? -ne 0 ]; then
+          echo "::error::Failed to check existing tickets: $response"
+          exit 1
+        fi
+
+        check_if_ticket_exists=$(echo "$response" | jq .issues[0].key)
+        if [[ "$check_if_ticket_exists" != "null" ]]; then
+          echo "abort_ticket_creation=true" >> $GITHUB_ENV
+          echo "::error::ticket found as $check_if_ticket_exists aborting ticket creation"
+        fi
+      shell: bash
+
+    - name: Create Jira Issue
+      if: ${{ env.abort_ticket_creation != 'true' }}
+      run: |
+        if [[ "${{ inputs.reason_for_ticket_creation }}" == "veracode" ]]; then
+          ticket_description=$(cat <<-EOD
+          {"version": 1, "type": "doc", "content": [
+            {"type": "paragraph", "content": [
+              {"type": "text", "text": "This incident ticket relates to:"}
+            ]},
+            {"type": "bulletList", "content": [
+                {"type": "listItem", "content": [
+                  {"type": "paragraph", "content": [
+                    {"type": "text", "text": "Ref: "},
+                    {"type": "text", "text": "PR-${{ github.event.pull_request.number }}", "marks": [{"type": "link", "attrs": {"href": "${{ github.event.pull_request.html_url }}"}}]}
+                  ]
+                }]
+              }]
+            },
+            {"type": "paragraph", "content": []},
+              {"type": "bulletList", "content": [
+                {"type": "listItem", "content": [
+                  {"type": "paragraph", "content": [
+                    {"type": "text", "text": "Target: "},
+                    {"type": "text", "text": "${{ github.base_ref || github.ref_name }}", "marks": [{"type": "strong"}]}
+                  ]}
+                ]},
+                {"type": "listItem", "content": [
+                  {"type": "paragraph", "content": [
+                    {"type": "text", "text": "Link to the job", "marks": [{"type": "link", "attrs": {"href": "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}"}}]}
+                  ]}
+                ]}
+              ]},
+            ]}
+        EOD
+          )
+        elif [[ "${{ inputs.reason_for_ticket_creation }}" == "nightly" ]]; then
+          ticket_description=$(cat <<-EOD
+          {"version": 1, "type": "doc", "content": [
+            {"type": "paragraph", "content": [
+              {"type": "text", "text": "This incident ticket relates to the ${{ inputs.module_name }} nightly on ${{ github.ref_name }} which failed on $(date '+%Y-%m-%d')."}
+            ]},
+            {"type": "paragraph","content": [
+              {"type": "text", "text": "Link to the failed nightly", "marks": [{"type": "link", "attrs": {"href": ${{ steps.get_context.outputs.run_url }} }}]}
+            ]}
+          ]}
+        EOD
+          )
+        fi
```

Comment:
the gsub("\\$\\{([^}]+)\\}"; . as $in | $in[$1] // "") is complete gibberish, but otherwise the idea isn't too bad 
do you have any better implementation of this that actually works and that takes into account the latest changes on the pull request?